src/utils.cpp

```diff
@@ -438,3 +439,33 @@ std::vector<std::string> Utils::Tokenize(const std::string &str_to_tokenize, con
 
 	return tokens;
 }
+
+
+std::string Utils::ReplacePlaceholders(const std::string& text_template, std::vector<char> types, std::vector<std::string> values) {
+	std::string str = text_template;
```

Comment:
Testcase:
%S appeared! %A %A %s%%s%

(Used %A because %O is substituted to ??? >.>)

2k result: Slime appeared! %A %A Slime%Slime%
EasyRPG: Slime appeared! %A %A Slime%%s%